src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -5881,12 +5860,17 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
       // Extract and insert stabilizing ValDefs (if any) which might have been
       // introduced during the typing of the original expression.
       def insertStabilizer(tree: Tree, original: Tree): Tree = {
-        if (phase.erasedTypes) tree
-        else stabilizingDefinitions(original) match {
-          case Nil => tree
-          case vdefs =>
-            removeStabilizingDefinitions(tree)
-            Block(vdefs.reverse, tree) setType tree.tpe setPos tree.pos
+        def definesStabilizer(sym: Symbol) = original.exists({
+          case sel: Select => sel.attachments.get[StabilizedQualifier].exists(_.sym == sym)
+          case _ => false
+        })
```

Comment:
IMOO (in my objective opinion) the extra parens on method call are unnecessary rather than :sparkles: extra :sparkles:.